src/plugin/track/confirmtrack.js

```diff
@@ -1,101 +1,108 @@
-  return {
-    restrict: 'E',
-    templateUrl: os.ROOT + 'views/plugin/track/confirmtrack.html',
-    controller: plugin.track.ConfirmTrackCtrl,
-    controllerAs: 'confirm'
-  };
-};
+const directive = () => ({
+  restrict: 'E',
+  templateUrl: os.ROOT + 'views/plugin/track/confirmtrack.html',
+  controller: Controller,
+  controllerAs: 'confirm'
+});
 
 
 /**
  * Add the directive to the coreui module
  */
-os.ui.Module.directive('confirmtrack', [plugin.track.confirmTrackDirective]);
+Module.directive('confirmtrack', [directive]);
 
 
 
 /**
  * Controller for the track selection window.
- *
- * @param {!angular.Scope} $scope
- * @constructor
- * @ngInject
+ * @unrestricted
  */
-plugin.track.ConfirmTrackCtrl = function($scope) {
-  var trackNode = plugin.places.PlacesManager.getInstance().getPlacesRoot();
-
+class Controller {
   /**
-   * The available tracks.
-   * @type {!Array<!ol.Feature>}
+   * Constructor.
+   * @param {!angular.Scope} $scope
+   * @ngInject
    */
-  this['tracks'] = trackNode ? trackNode.getFeatures() : [];
+  constructor($scope) {
+    var trackNode = PlacesManager.getInstance().getPlacesRoot();
+
+    /**
+     * The available tracks.
+     * @type {!Array<!OlFeature>}
+     */
+    this['tracks'] = trackNode ? trackNode.getFeatures() : [];
+
+    /**
+     * The selected track.
+     * @type {OlFeature|undefined}
+     */
+    this['track'] = this['tracks'][0] || undefined;
+
+    $scope.$watch('confirm.track', function(newVal, oldVal) {
+      $scope.$parent['confirmValue'] = newVal;
+    });
+
+    $scope.$emit(WindowEventType.READY);
+  }
 
   /**
-   * The selected track.
-   * @type {ol.Feature|undefined}
+   * Get the name of a track.
+   *
+   * @param {!OlFeature} track The track
+   * @return {string} The name
+   * @export
    */
-  this['track'] = this['tracks'][0] || undefined;
-
-  $scope.$watch('confirm.track', function(newVal, oldVal) {
-    $scope.$parent['confirmValue'] = newVal;
-  });
-
-  $scope.$emit(os.ui.WindowEventType.READY);
-};
-
+  getTrackName(track) {
+    var trackName = /** @type {string|undefined} */ (track.get(KMLField.NAME));
+    if (!trackName) {
+      return 'Unnamed Track (id =' + track.getId() + ')';
+    }
 
-/**
- * Get the name of a track.
- *
- * @param {!ol.Feature} track The track
- * @return {string} The name
- * @export
- */
-plugin.track.ConfirmTrackCtrl.prototype.getTrackName = function(track) {
-  var trackName = /** @type {string|undefined} */ (track.get(plugin.file.kml.KMLField.NAME));
-  if (!trackName) {
-    return 'Unnamed Track (id =' + track.getId() + ')';
+    return trackName;
   }
-
-  return trackName;
-};
-
+}
 
 /**
  * Prompt the user to choose a track.
  *
- * @return {!goog.Promise}
+ * @return {!Promise}
  */
-plugin.track.promptForTrack = function() {
-  return new goog.Promise(function(resolve, reject) {
-    plugin.track.launchConfirmTrack(resolve, reject);
+pluginTrack.promptForTrack = function() {
```

Comment:
The `plugin.track` functions in this file will need to be refactored given this is not the `plugin.track` module. They'll either need to be moved, or exported by this module and references updated.

Reply:
Moved over to TrackManager without causing any circular dependencies. Updated for the breaking change downstream.